src/events.zig

```diff
@@ -439,6 +442,48 @@ pub const Key = struct {
     repeat: bool,
 };
 
+/// Mouse button event structure
+///
+/// ## Version
+/// This struct is available since SDL 3.2.0.
+///
+/// ## Code Examples
+/// ```zig
+///                 see https://wiki.libsdl.org/SDL3/SDL_MouseButtonFlags
+///                                                              left button
+/// if (event.* == .mouse_button_down and event.mouse_button_down.button == 1) {
+///     pressing = true;
+/// }
+///                                                              left button
+/// if (event.* == .mouse_button_up and event.mouse_button_up.button == 1) {
+///     pressing = false;
+/// }
```

Comment:
I feel like using a switch statement with the tagged union of `event` would more zig-like code.